### DISCUSSION TITLE
Working with TypeORM

### DISCUSSION THREAD

**Discussion by muhaimincs** [planetscale/discussion#71] [FAQs] [answered]
I want to use PlanetScale with existing project. TypeORM already has it's own mechanism to connect db as well as it's own query. How do I resolve it

**Comment by jgreet** [comment]
Hi @muhaimincs, [TypeORM supports MySQL](https://typeorm.io/#/connection), and PlanetScale is built on MySQL 8, so you should be able to connect by running the [pscale connect](https://docs.planetscale.com/tutorial/connect-any-application) proxy and configuring TypeORM to use the mysql adapter:

```bash
$ pscale connect you_db_name branch_name
```

In ormconfig.json:
```js
{
   "type": "mysql",
   "host": "localhost",
   "port": 3306,
   "username": "root", 
   "password": "password",    // unused but may need to be present
   "database": "your_db_name"
}
 ```

When connection strings are available they should be an option for configuring TypeORM apps as well. See https://github.com/planetscale/beta/discussions/15.

**Reply by muhaimincs** [reply]
I can't find ways to connect to db on vercel as they use token. I've tried to connect via localhost with root as username and no password but no luck. Would you enlighten a bit more.

I don't have problem on development env to connect via localhost

**Reply by jgreet** [reply]
Unfortunately if you're hosting the app on Vercel this approach won't work. There's no way to start the `pscale` proxy in that environment.

Since we don't have a compatible driver for TypeORM, I think your best bet is to wait for connection strings to be available (https://github.com/planetscale/beta/discussions/15#discussioncomment-1061800) at which point you should be able to use TypeORM's `mysql` driver.

**Reply by muhaimincs** [reply]
ok noted

**Comment by mscoutermarsh** [comment, marked as answer]
Update! You can now [generate a connection string](https://docs.planetscale.com/concepts/connection-strings), then establish a normal MySQL connection with TypeORM.

**Reply by muhaimincs** [reply]
definitely will give it a go

**Reply by muhaimincs** [reply]
I'm having problem with TypeORM running on Vercel

```js
const ssl = process.env.NODE_ENV !== "production" ? true : {
  ca : process.env.PLANETSCALE_SSL_CERT_PATH,
}

const options: any = {
    ...connectionOptions,
    entities: [Category, Author, Post, ListMeta, PostCategory],
    migrations: [__dirname + '/migrations/*.ts'],
    dateStrings: true,
    type: 'mysql',
    timezone: 'Z',
    // logging: true,
    // debug: true,
    ssl,
    ...optionOverrides
  };
```

this will work on `localhost` but no on Vercel. And it end up with `No connection options were found in any orm configuration files.`. Anyone use this config on Vercel?
